lib/ansible/modules/cloud/azure/azure_rm_keyvault.py

```diff
@@ -0,0 +1,436 @@
+                if update_tags:
+                    changed = True
+
+            elif self.state == 'absent':
+                changed = True
+
+        except CloudError:
+            # the vault does not exist (it's deleted) so create it
+            if self.state == 'present':
+                changed = True
+
+            else:
+                # you can't delete what is not there
+                changed = False
+
+        self.results['changed'] = changed
+        self.results['state'] = results
+
+        if self.tenant_id is None:
+            self.tenant_id = self.credentials['tenant']
+        if self.object_id is None:
+            self.object_id = self.credentials['client_id']
+
+        # return the results if your only gathering information
+        if self.check_mode:
+            return self.results
+
+        if changed:
+            if self.state == 'present':
+                # if you want to create or update a key vault
+                if not vault:
+                    # create new vault
+                    self.log('Creating vault {0}'.format(self.name))
+                    use_sku = Sku(self.sku)
+                    permission_preset = create_permissions(self)
+                    access_policy_lst = [AccessPolicyEntry(self.tenant_id, self.object_id, permission_preset, application_id=self.application_id)]
+                    vault_properties = VaultProperties(self.tenant_id,
+                                                       use_sku,
+                                                       access_policies=access_policy_lst,
+                                                       vault_uri=self.vault_uri,
+                                                       enabled_for_deployment=self.enabled_for_deployment,
+                                                       enabled_for_disk_encryption=self.enabled_for_disk_encryption,
+                                                       enabled_for_template_deployment=self.enabled_for_template_deployment)
+                    vault = VaultCreateOrUpdateParameters(self.location, vault_properties, tags=self.tags)
+                else:
+                    # update the vault
+                    vault = VaultCreateOrUpdateParameters(self.location, results['properties'], tags=results['tags'])
+
+                self.results['state'] = self.create_or_update_vault(vault)
+            elif self.state == 'absent':
+                # delete vault
+                self.delete_vault()
+                # the delete does not actually return anything. if no exception, then we'll assume
+                # it worked.
+                self.results['state']['status'] = 'Deleted'
+        return self.results
+
+    def create_or_update_vault(self, vault):
+        try:
+            # create or update the new Vault object we created
+            new_vault = self.keyvault_client.vaults.create_or_update(self.resource_group, self.name, vault)
```

Comment:
define a `poller` and `get_poller_result(poller)` as response

Reply:
@yuwzho do we have an existing example?

Reply:
@yuwzho I found an example in VM module.

Reply:
The return type of create_or_update is vault and not a AzureOperationPoller like other types (i.e. VM). We would need to create a polling operation for this, but I feel it's not needed. The response is returned by the API after the vault has been created and the code path continues to execute.